packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -197,7 +197,8 @@ export class ElectronMainApplication {
     }
 
     async start(config: FrontendApplicationConfig): Promise<void> {
-        this.useNativeWindowFrame = this.getTitleBarStyle(config) === 'native';
+        const args = this.processArgv.getProcessArgvWithoutBin(process.argv);
+        this.useNativeWindowFrame = this.getTitleBarStyle(config) === 'native' && !args.includes('--no-native-window-frame');
```

Comment:
Would it make sense to make this a theia `CliContribution` that we can include into apps to be tested?